pkg/prober/prober.go

```diff
@@ -0,0 +1,155 @@
+package prober
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/common/route"
+)
+
+const (
+	healthyEndpointPath  = "/-/healthy"
+	readyEndpointPath    = "/-/ready"
+	okProbeText          = "thanos %v is %v"
+	errorProbeText       = "thanos %v is not %v. Reason: %v"
+	probeErrorHTTPStatus = 500
+	initialErrorText     = "thanos %s is initializing"
+)
+
+// Prober represents health and readriness status of given compoent.
+type Prober struct {
+	logger       log.Logger
+	loggerMtx    sync.Mutex
+	componentMtx sync.Mutex
+	component    string
```

Comment:
Why not use the native component struct ?

Reply:
Ha, I remember thinking the same for myself before I forget about it again :smile:
Thanks, I'll change it